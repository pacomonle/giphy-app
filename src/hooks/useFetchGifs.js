import { useEffect, useState } from "react"
import getGifs from '../helpers/getGifs';

export const useFetchGifs = (category) => {

    const [state, setstate] = useState({
        data:[],
        loading: true
    })
    
            useEffect(() => {
                getGifs(category).then(
                res =>{
                   
                        setstate({
                            data: res,
                            loading: false
                        })
               
                })
            }, [category])
       
   
    return state
}
