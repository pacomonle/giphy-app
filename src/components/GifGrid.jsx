import React /*,{ useEffect, useState }*/ from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem'
// import getGifs from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({category}) => {
/*
   const [images, setimages] = useState([])

    useEffect(() => {

     getGifs(category).then(
         res =>{
            setimages( res )
         })

    }, [category])
S
  */
 // renombrando en la destructuracion con :
  const {loading, data:images} = useFetchGifs(category)
  

    return (
        <>
            <h3 className='animate__animated animate__flipInX'>{category}</h3>

            {
                loading && (<h3 className='animate__animated animate__flash'>Cargando...</h3>)                
            }
            <div className='card-grid'>
                {
                images.map( item  => (
                    <GifGridItem key={item.id} {...item}></GifGridItem>        
                ))  
                }
            </div>
            
                
          
        </>
    )
}

GifGrid.propTypes = {
  category : PropTypes.string.isRequired
}

export default GifGrid

