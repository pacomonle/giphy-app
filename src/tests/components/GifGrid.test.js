import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid></GifGrid>', () => {

    const category = 'Hola Mundo'

    test('should <GifGrid /> montado correctamete', () => {
        // simulando el useFecthGifs
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true            
        })
        const wraper = shallow(<GifGrid category = {category}></GifGrid>)
        expect(wraper).toMatchSnapshot()
    })
    
    test('should mostrar items al cargar imagenes', () => {
        // simular respuesta
        const gifs =[{
            id: 'ABC',
            url: 'https://localhost8080/any/cosa.jpg',
            title: 'Cualquier cosa'
        }]
        // a√±adir respuesta al useFetchGifs
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false   
        })
        // comprobar snapShot
        const wraper = shallow(<GifGrid category = {category}></GifGrid>)
        expect(wraper).toMatchSnapshot()
        // evaluando el { loading && <p> className= ' animate_animated animate_flash' > Loading </p> }
        expect(wraper.find('p').exists()).toBe(false)
        // evaluando si existe el componente <GifGridItem></GifGridItem>
        console.log(wraper.find('GifGridItem').props())
        expect(wraper.find('GifGridItem').length).toBe(gifs.length)
       

    })
    
})
