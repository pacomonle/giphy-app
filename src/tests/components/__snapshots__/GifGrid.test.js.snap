// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid></GifGrid> should <GifGrid /> montado correctamete 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    Hola Mundo
  </h3>
  <h3
    className="animate__animated animate__flash"
  >
    Cargando...
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid></GifGrid> should mostrar items al cargar imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    Hola Mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost8080/any/cosa.jpg"
    />
  </div>
</Fragment>
`;
